modules/caddypki/ca.go

```diff
@@ -75,10 +75,11 @@ type CA struct {
 	// and module provisioning.
 	ID string `json:"-"`
 
-	storage     certmagic.Storage
-	root, inter *x509.Certificate
-	interKey    any // TODO: should we just store these as crypto.Signer?
-	mu          *sync.RWMutex
+	storage    certmagic.Storage
+	root       *x509.Certificate
+	interChain []*x509.Certificate
+	interKey   any // TODO: should we just store these as crypto.Signer?
```

Comment:
While we're here, curious what your thoughts are on using crypto.Signer -- do you typically use that interface, or `any`? I feel like I saw only `any` back in the day for private keys, but maybe `crypto.Signer` is more specific?

Reply:
Oh... you know, that would exclude `ecdh.PrivateKey` -- that is probably fine, since this is intended for DH key exchange... so yeah, if that's okay, and you don't mind while you're at it, let's switch to `crypto.Signer` :)

Thanks!